prow/cmd/crier/main.go

```diff
@@ -0,0 +1,164 @@
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"os"
+	"os/signal"
+	"syscall"
+
+	"github.com/sirupsen/logrus"
+	meta_v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/util/workqueue"
+
+	prowjobclientset "k8s.io/test-infra/prow/client/clientset/versioned"
+	prowjobinformer "k8s.io/test-infra/prow/client/informers/externalversions/prowjobs/v1"
+)
+
+// TODO(krzyzacy): copy & paste, refactor this
+// loadClusterConfig loads connection configuration
+// for the cluster we're deploying to. We prefer to
+// use in-cluster configuration if possible, but will
+// fall back to using default rules otherwise.
+func loadClusterConfig() (*rest.Config, error) {
+	clusterConfig, err := rest.InClusterConfig()
+	if err == nil {
+		return clusterConfig, nil
+	}
+
+	credentials, err := clientcmd.NewDefaultClientConfigLoadingRules().Load()
+	if err != nil {
+		return nil, fmt.Errorf("could not load credentials from config: %v", err)
+	}
+
+	clusterConfig, err = clientcmd.NewDefaultClientConfig(*credentials, &clientcmd.ConfigOverrides{}).ClientConfig()
+	if err != nil {
+		return nil, fmt.Errorf("could not load client configuration: %v", err)
+	}
+	return clusterConfig, nil
+}
+
+// TODO(krzyzacy): move this to kube???
+// getKubernetesClient retrieves the Kubernetes cluster
+// client from within the cluster
+func getKubernetesClient() (kubernetes.Interface, prowjobclientset.Interface) {
+	config, err := loadClusterConfig()
+	if err != nil {
+		return fmt.Errorf("failed to load cluster config: %v", err)
+	}
+
+	// generate the client based off of the config
+	client, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		logrus.Fatalf("getClusterConfig: %v", err)
+	}
+
+	prowjobClient, err := prowjobclientset.NewForConfig(config)
+	if err != nil {
+		logrus.Fatalf("getClusterConfig: %v", err)
+	}
+
+	logrus.Info("Successfully constructed k8s client")
+	return client, prowjobClient
+}
+
+func main() {
+	logrus.SetFormatter(
+		logrusutil.NewDefaultFieldsFormatter(nil, logrus.Fields{"component": "crier"}),
+	)
+
+	// get the Kubernetes client for connectivity
+	client, prowjobClient := getKubernetesClient()
+
+	// retrieve our custom resource informer which was generated from
+	// the code generator and pass it the custom resource client, specifying
+	// we should be looking through all namespaces for listing and watching
+	informer := prowjobinformer.NewProwJobInformer(
```

Comment:
You should use factories for the informers so they can share caches

Reply:
sure